cli/flox-rust-sdk/src/models/manifest/composite.rs

```diff
@@ -0,0 +1,276 @@
+#![allow(dead_code)] // TODO: Remove on first use.
+
+use std::iter::once;
+
+use flox_core::Version;
+use thiserror::Error;
+
+use super::typed::{
+    Manifest,
+    ManifestBuild,
+    ManifestContainerize,
+    ManifestHook,
+    ManifestInstall,
+    ManifestOptions,
+    ManifestProfile,
+    ManifestServices,
+    ManifestVariables,
+};
+
+#[derive(Error, Debug)]
+pub enum MergeError {}
+
+/// A collection of manifests to be merged with a `ManifestMergeStrategy`.
+struct CompositeManifest {
```

Comment:
Suggestion: I don't think I would call this `*Manifest` since it's not itself a manifest. It's just a container for manifests that are about to be merged.

Reply:
`MergeNode` makes sense. `Composite` is too general, @mkenigs calls this container a `RequiresNode`, which I think is tied too closely to the actual manifest format, and `MergeNode` tells you exactly what it is, a node in the merge graph.

Reply:
I'm not sure about `MergeNode` because the container represents all of the nodes rather than one/two (we're not doing recursion - I don't feel like there's any benefit right now).

Reply:
`Composer`, as the container that is "composing" and not actually representing the compos*ed* manifest (i.e the result of `XYZ::merge_all`)

Reply:
Then we stutter `Composer.composer` unless we find another name for the composer manifest like `parent` 🚲